cloud/pod_util.go

```diff
@@ -95,6 +100,49 @@ func ExportPodResources(info pod.ResourceInfo) cocoa.ECSPodResources {
 	return *res
 }
 
+// ExportPodContainerDef exports the ECS pod container definition into the equivalent cocoa.ECSContainerDefintion.
+func ExportPodContainerDef(opts pod.TaskContainerCreationOptions) (*cocoa.ECSContainerDefinition, error) {
+	return cocoa.NewECSContainerDefinition().
+		SetImage(opts.Image).
+		SetMemoryMB(opts.MemoryMB).
+		SetCPU(opts.CPU).
+		SetEnvironmentVariables(exportEnvVars(opts.EnvVars, opts.EnvSecrets)), nil
+}
+
+// ExportPodExecutionOptions exports the ECS configuration into cocoa.ECSPodExecutionOptions.
+func ExportPodExecutionOptions(ecsConfig evergreen.ECSConfig, podOS pod.OS) (*cocoa.ECSPodExecutionOptions, error) {
+	if len(ecsConfig.Clusters) == 0 {
+		return nil, errors.New("must specify at least one cluster to use")
+	}
+
+	for _, cluster := range ecsConfig.Clusters {
+		if string(podOS) == string(cluster.Platform) {
+			return cocoa.NewECSPodExecutionOptions().SetCluster(cluster.Name), nil
+		}
+	}
+
+	return nil, errors.Errorf("cluster platform ('%s') and pod OS ('%s') must match", string(podOS), string(ecsConfig.Clusters[0].Platform))
```

Comment:
```go
errors.Errorf("pod OS '%s' did not match any ECS cluster platforms", string(podOS))
```

Reply:
Agreed that this is a better way to error, though I think the check on L114 (`len(cluster) == 0`) would prevent the program from panicking if there were no clusters.

Reply:
True. The length check is not necessary since for-ranging over a nil/empty slice will no-op, but it's not wrong to leave it in.